### PR TITLE
git-lfs: use buildGoModule

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by booklearner** [comment]
Result of `nixpkgs-review pr 176180` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>crystal2nix</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>crate2nix</li>
    <li>dep2nix</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>go2nix</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update-source</li>
    <li>npins</li>
    <li>prefetch-yarn-deps</li>
    <li>update-nix-fetchgit</li>
    <li>vgo2nix</li>
    <li>yarn2nix</li>
  </ul>
</details>

**Review by booklearner — APPROVED** [review]

File: pkgs/applications/version-management/git-lfs/default.nix [hunk 1]
```diff
@@ -1,43 +1,56 @@
...
+  ];
 
   subPackages = [ "." ];
 
   preBuild = ''
-    pushd go/src/github.com/git-lfs/git-lfs
-      go generate ./commands
-    popd
+    go generate ./commands
   '';
```
Comment (marsam) [inline pkgs/applications/version-management/git-lfs/default.nix:29]: Please remove this. IIUC is not necessary, it regenerates the .ronn documents, and it should be handled upstream.

Comment (marsam) [reply pkgs/applications/version-management/git-lfs/default.nix:29]: I'm so sorry, my bad.
 This command is necessary for injecting help information into the commands. The same it's applied in ArchLinux https://github.com/archlinux/svntogit-community/commit/476c403f4cbfcb29e84a47130cef1d4797a09057. 
I'm reverting it to your original implementation. Thanks!

**Review by marsam — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**